config/initializers/timeout.rb

```diff
@@ -1,7 +1,7 @@
 # License: AGPL-3.0-or-later WITH Web-Template-Output-Additional-Permission-3.0-or-later
-timeout = Integer(ENV['WEB_TIMEOUT'] || 15)
-if ENV['RAILS_ENV'] == 'development' || ENV['IDE_PROCESS_DISPATCHER']
-  timeout = 10000
-end
 
-Rack::Timeout.timeout = timeout  # seconds
\ No newline at end of file
+# set the Rack timeout to 2 hours when running in development so we can use breakpoints
+# For staging and prod, you should set the RACK_TIMEOUT_SERVICE_TIMEOUT environment variable
+if Rails.env.development? || ENV["IDE_PROCESS_DISPATCHER"]
+  ENV["RACK_TIMEOUT_SERVICE_TIMEOUT"] = 2.hours.to_s
+end
```

Comment:
So you no longer need to set the actual timeout? 

```Rack::Timeout.timeout = something...```

And that happens via whatever `ENV["RACK_TIMEOUT_SERVICE_TIMEOUT"]` is set to?

Reply:
Good question: it defaults to 15 seconds if that environment variable isn't set.